ocaml/xapi/cpuid_helpers.ml

```diff
@@ -45,41 +45,27 @@ let zero_extend arr len =
   let zero_arr = Array.make len 0L in
   extend arr zero_arr
 
-let features_op2 f a b =
-  let n = max (Array.length a) (Array.length b) in
-  Array.map2 f (zero_extend a n) (zero_extend b n)
-
-(** Calculate the intersection of two feature sets.
- *  Intersection with the empty set is treated as identity, so that intersection
- *  can be folded easily starting with an accumulator of [||].
- *  If both sets are non-empty and of differing lengths, and one set is longer
- *  than the other, the shorter one is zero-extended to match it.
- *  The returned set is the same length as the longer of the two arguments.  *)
-let intersect left right =
-  match left, right with
-  | [| |], _ -> right
-  | _, [| |] -> left
-  | _, _ -> features_op2 Int64.logand left right
-
-(** Calculate the features that are missing from [right],
-  * but present in [left] *)
-let diff left right =
-  let diff64 a b = Int64.(logand a (lognot b)) in
-  features_op2 diff64 left right
-
-(** equality check that zero-extends if lengths differ *)
-let is_equal left right =
-  let len = max (Array.length left) (Array.length right) in
-  (zero_extend left len) = (zero_extend right len)
-
-(** is_subset left right returns true if left is a subset of right *)
-let is_subset left right =
-  is_equal (intersect left right) left
-
-(** is_strict_subset left right returns true if left is a strict subset of right
-    (left is a subset of right, but left and right are not equal) *)
-let is_strict_subset left right =
-  (is_subset left right) && (not (is_equal left right))
+let intersect ~__context left right =
+  let open Xapi_xenops_queue in
+  let dbg = Context.string_of_task __context in
+  let module Client = (val make_client (default_xenopsd ()): XENOPS) in
```

Comment:
you could create this module once in the global scope of this file? or does making the client have a side effect?